### ISSUE TITLE
 Is there any mismatch for hardswish_backend when input x equal to 3

### ISSUE THREAD

**Issue by i4oolish** [pytorch/pytorch#87778]
### 🐛 Describe the bug

From the define in the pytorch document [link](https://pytorch.org/docs/stable/generated/torch.nn.Hardswish.html?highlight=hardswish#torch.nn.Hardswish), it seems the `d_in = dout * 1` when `x = 3`, while from the [implementation](https://github.com/pytorch/pytorch/pull/34747/files), we can see `d_in = d_out * (1+0.5)`, is there any mistake between the implementation and the defination?

[1]: https://pytorch.org/docs/stable/generated/torch.nn.Hardswish.html?highlight=hardswish#torch.nn.Hardswish
[2]: https://github.com/pytorch/pytorch/pull/34747/files

### Versions

Collecting environment information...
PyTorch version: 1.10.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.25

Python version: 3.6.9 (default, Jun 29 2022, 11:45:57)  [GCC 8.4.0] (64-bit runtime)
Python platform: Linux-4.15.0-142-generic-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.10.0
[pip3] torchaudio==0.10.0
[pip3] torchvision==0.11.1
[conda] Could not collect

cc @svekars @carljparker @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

**Comment by bdhirsh** [comment]
The docs don't technically say anything about the backward formula, although maybe the extra 0.5 has something to do with being at the boundary of the piece-wise function? If that's the case, maybe we could update the docs

**Comment by i4oolish** [comment]
> The docs don't technically say anything about the backward formula, although maybe the extra 0.5 has something to do with being at the boundary of the piece-wise function? If that's the case, maybe we could update the docs

Thanks for your reply. The background of this issue is when I try to implement hard_swish_backward based on the docs, I found the training accuracy based on our AI software on our DSA hw, is not as higher as PyTorch on GPU. Then I doubt maybe the extra 0.5 have some effect about the training accuracy.
After I modify the hard_swish_backward to keep same with PyTorch implementation, the training accuracy have a little improvement.
